server.py

```diff
@@ -345,6 +345,13 @@ def main():
         quit(1)
     port = sys.argv[1]
 
+    databasePassword = input("Input database password: ")
+    try:
+        global database
```

Comment:
You are declaring `database` to be global here and also declaring it on line 25.  I am not sure exactly how python handles this but it might led to some problems.  Maybe just declare it as global here and remove it from line 25.

Reply:
From what I understand it needs to be on line 25 and here. By default, if there is a local decleration by the same name as a global one ('database') then it will stay in local scope. By saying `global database` I am telling python not to create a new local variable database, but instead to reference the existing global one